### PR TITLE
Nj 203 signature date arin

### PR DESCRIPTION
**Pull request by arinchoi03**
## Link to pivotal/JIRA issue
- https://github.com/codeforamerica/vita-min/pull/5264 following up with a different testing approach to be least disruptive; will have a few followups as a result of this, potentially
## Is PM acceptance required? (delete one)
- Yes - on the original ticket though, not this PR

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Reverted the last commit to make changes to the factory bots 
- Moved setting of `_esigned_at` and `esigned` setting to the test itself (not in factory) -- ideally I think we should have a trait that sets primary/secondary esigned 
- going to attempt setting the base nc fixture/state to a single status one, but may split that into its own work if it proves cumbersome

## How to test?
- Describe the testing approach taken to verify the changes, including:
  - Unit/integration/manual tests
  - Test data used
- Specify any relevant testing environments used (e.g., development, staging, demo, Heroku).
- Risk Assessment
  - Just making sure the calculations that are changed in the tests due to using a new fixture makes sense.
## Screenshots (for visual changes)
- Before
- After

### CODE REVIEW

File: spec/factories/state_file_az_intakes.rb [hunk 1]
```diff
@@ -232,7 +228,7 @@
     end
 
     factory :state_file_az_refund_intake do
-      after(:build) do |intake, _evaluator|
+      after(:build) do |intake, evaluator|
```
Comment (arinchoi03) [inline spec/factories/state_file_az_intakes.rb:231]: was this added by your editor/rubocop? I think we can add it back

File: spec/factories/state_file_md_intakes.rb [hunk 2]
```diff
@@ -129,10 +129,9 @@
     primary_first_name { "Mary" }
     primary_middle_initial { "A" }
     primary_last_name { "Lando" }
-    primary_birth_date { Date.new(1950, 0o1, 0o1) } # matches the bday in md_minimal.json
+    primary_birth_date { Date.new(1950, 01, 01) } # matches the bday in md_minimal.json
     primary_signature_pin { '12345' }
-    primary_esigned_at { DateTime.new(2024, 12, 19, 12)}
-    primary_esigned { 'yes' }
+    primary_esigned_at { DateTime.now }
```
Comment (arinchoi03) [inline spec/factories/state_file_md_intakes.rb:134]: i'll probably just remove this & add this to the spec that needed it

File: spec/factories/state_file_nc_intakes.rb [hunk 3]
```diff
@@ -87,7 +87,7 @@
 FactoryBot.define do
   factory :state_file_nc_intake do
     transient do
-      filing_status { 'single' }
+      filing_status { 'married_filing_jointly' }
```
Comment (arinchoi03) [inline spec/factories/state_file_nc_intakes.rb:90]: ill probably work on this a bit in this PR

File: spec/lib/pdf_filler/nc_d400_pdf_spec.rb [hunk 4]
```diff
@@ -29,6 +30,10 @@
                primary_esigned_at: tomorrow_midnight)
       }
 
+      before do
+        intake.direct_file_data.mailing_apartment = mailing_apartment
+      end
```
Comment (arinchoi03) [inline spec/lib/pdf_filler/nc_d400_pdf_spec.rb:35]: daffy does not have a apartment number, but wanted to add it to make sure it was getting filled in

File: spec/factories/state_file_az_intakes.rb (hunk 1 above)
Comment (jachan) [reply spec/factories/state_file_az_intakes.rb:231]: It was an automatic rubocop change on save. 

> Lint/UnusedBlockArgument: Unused block argument - `evaluator`. If it's necessary, use `_` or `_evaluator` as an argument name to indicate that it won't be used.RuboCop(Lint/UnusedBlockArgument)

File: spec/factories/state_file_md_intakes.rb (hunk 2 above)
Comment (jachan) [reply spec/factories/state_file_md_intakes.rb]: That sounds good

**Comment by jachan** [comment]
Closing and reopening this PR to retrigger Heroku build

**Pull request merged**